provider/azure/environ.go

```diff
@@ -262,7 +263,7 @@ func (env *azureEnviron) Bootstrap(
 		wg.Add(1)
 		go func(id string) {
 			defer wg.Done()
-			sdkCtx := stdcontext.Background()
+			sdkCtx := context.Background()
```

Comment:
We now have a context that we're passing in via the Bootstrap method args, so we need to now use that one and now create a separate one here, ie we want this azure bootstrap process to react to the passed in ctx getting cancelled.
And as I said elsewhere, adding Context() to BootstrapContext means we don't need to change the Bootstrap() method signature at all, we just do ctx.Context() instead of stdcontext.Background()

Reply:
I think this should actually be a separate context, as it's cleanup and shouldn't be cancelled if the bootstrap is cancelled (the cleanup *should* be run if the bootstrap is cancelled).